### PR TITLE
Authenticate for public uploads and support private uploads without `ask`ing

### PR DESCRIPTION
**Pull request by goto-bus-stop**
`clinic upload xyz.clinic-doctor` now also requires authentication.

A new `clinic upload --private` flag uploads to your private area.
- [x] This needs a test, can just copy and slightly modify one from `ask`

`clinic ask` does `clinic upload --private` and then calls a currently-noop function that can be implemented once we have a `/ask` endpoint on the server.

### CODE REVIEW

File: bin.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ const fs = require('fs')
 const path = require('path')
 const opn = require('opn')
 const ora = require('ora')
-const async = require('async')
```
Comment (mcollina) [inline bin.js]: are we using `async` in some other place? Should we remove it from our deps?

File: bin.js [hunk 2]
```diff
@@ -52,52 +52,36 @@ const result = commist()
...
-        'upload-url': 'https://upload.clinicjs.org'
+        'upload-url': DEFAULT_UPLOAD_URL
       }
     })
 
     if (args.help) {
       printHelp('clinic-upload')
     } else if (args._.length > 0) {
-      checkMetricsPermission(() => {
+      checkMetricsPermission(async () => {
```
Comment (mcollina) [inline bin.js]: Can you add a comment explaining that `checkMetricsPermission` does not add a catch handler, and as such we are wrapping all the content of this function in a try/catch? Can you also move the `trackEvent` call inside the try-catch?

File: docs/clinic-upload.txt [hunk 3]
```diff
@@ -14,5 +14,11 @@
 
   Replace 13242 with your pid.
 
+  If your profile contains potentially sensitive data, you can upload it
+  privately to your account using <code>--private</code>.
+
+    <code>clinic upload --private 13242.clinic-flame</code>
```
Comment (mcollina) [inline docs/clinic-upload.txt]: CLINIC_JWT should be documented here.

File: bin.js (hunk 1 above)
Comment (goto-bus-stop) [reply bin.js]: It's still in use in `lib/clean.js`

**Pull request merged**